doc/api/net.md

```diff
@@ -2,74 +2,133 @@
+
+The listener callback is called without passing any arguments.
+
+```js
+server.on('close', () => {
+  console.log('The server has been closed.');
+});
+```
 
 ### Event: 'connection'
 <!-- YAML
 added: v0.1.90
 -->
 
-* {net.Socket} The connection object
+The `'connection'` event is emitted when a new connection is established.
 
-Emitted when a new connection is made. `socket` is an instance of
-`net.Socket`.
+The listener callback is called with the `net.Socket` object for the connection
+passed as the only argument.
+
+```js
+server.on('connection', (socket) => {
+  console.log('A connection has been established');
+});
+```
 
 ### Event: 'error'
 <!-- YAML
 added: v0.1.90
 -->
 
-* {Error}
+The `'error'` event is emitted when an error occurs. The [`'close'`][] event
+will be emitted immediately following the `'error'` event.
+
+The listener callback is called with the `Error` object passed as the only
+argument.
 
-Emitted when an error occurs.  The [`'close'`][] event will be called directly
-following this event.  See example in discussion of `server.listen`.
+```js
+server.on('error', (error) => {
+  console.log('An error occurred!', error);
+});
+```
 
 ### Event: 'listening'
 <!-- YAML
 added: v0.1.90
 -->
 
-Emitted when the server has been bound after calling `server.listen`.
+The `'listening'` event is emitted when the server has been bound by calling the
+`server.listen()` method.
+
+The listener callback is called without passing any arguments:
+
+```js
+server.on('listening', () => {
+  console.log('The server is now listening for connections!');
+});
+```
 
 ### server.address()
 <!-- YAML
 added: v0.1.90
 -->
 
-Returns the bound address, the address family name, and port of the server
-as reported by the operating system.
-Useful to find which port was assigned when getting an OS-assigned address.
-Returns an object with `port`, `family`, and `address` properties:
-`{ port: 12346, family: 'IPv4', address: '127.0.0.1' }`
+The `server.address()` method returns the address to which the server is
+currently bound.
+
+If the server is bound to a network host and port, an object will be returned
+with the following properties:
 
-Example:
+* `port` {number}
+* `family` {String} Will be either `'IPv4'` or `'IPv6'`
+* `address` {String} The IPv4 or IPv6 network address
+
+For example:
 
 ```js
-var server = net.createServer((socket) => {
-  socket.end('goodbye\n');
-}).on('error', (err) => {
-  // handle errors here
-  throw err;
+{
+  port: 12346,
+  family: 'IPv4',
+  address: '127.0.0.1'
+}
+```
+
+If the server is bound to a path, that path will be returned as a string. For
+instance:
+
+```js
+const net = require('net');
+
+const myServer = net.createServer();
+
+myServer.listen('/tmp/path', () => {
+  console.log(myServer.address());
+    // Prints: /tmp/path
 });
+```
+
+The `server.address()` method should not be called until after the `listening`
+event has been emitted, as illustrated in the following example:
```

Comment:
the code below is obscure, because there is no reference to the `'listening'` event, except the indirect fact that .listen()'s callback gets added as a listener to it... better show direct usage.

Also, better to just document that until the server is listening, .address() will return `null`, and leave up to users if they want to call .address(). Calling is a valid way to see if the server is listening already, for example.